### PR TITLE
fix(m1) add /opt/homebrew prefix to path and cpath

### PR DESCRIPTION
**Pull request by gszr**
## Summary

Homebrew on Apple M1 uses the prefix `/opt/homebrew`. This PR ensures LUA_PATH and LUA_CPATH are set appropriately in such case. (On Intel, homebrew uses `/usr/local` as its default prefix; as LuaJIT includes `/usr/local/share/lua/5.1/` in its default list of Lua paths, the formula worked without the need for a patch.)

Additionally, the default Kong prefix is set to `HOMEBREW_PREFIX` rather than being hardcoded to `/usr/local/opt/kong/` (which isn't writeable by Brew on M1).

Fixes Kong/kong#7705.

### CODE REVIEW

File: Formula/kong.rb [hunk 1]
```diff
@@ -40,17 +40,42 @@ def install
...
 diff --git a/kong/templates/kong_defaults.lua b/kong/templates/kong_defaults.lua
-index e38b475..7a74a2f 100644
+index 5937dad10..c3387fded 100644
 --- a/kong/templates/kong_defaults.lua
 +++ b/kong/templates/kong_defaults.lua
 @@ -1,5 +1,5 @@
  return [[
 -prefix = /usr/local/kong/
-+prefix = /usr/local/opt/kong/
++prefix = HOMEBREW_PREFIX/opt/kong/
```
Comment (fffonion) [inline Formula/kong.rb:67]: this looks bit weird, should this be something like `(os.getenv("HOMEBREW_PREFIX") or "/usr/local") .. "/opt/kong"`?

File: Formula/kong.rb [hunk 2]
```diff
@@ -40,17 +40,42 @@ def install
...
  proxy_access_log = logs/access.log
  proxy_error_log = logs/error.log
+@@ -166,8 +166,8 @@ lua_socket_pool_size = 30
+ lua_ssl_trusted_certificate = NONE
+ lua_ssl_verify_depth = 1
+ lua_ssl_protocols = TLSv1.1 TLSv1.2 TLSv1.3
+-lua_package_path = ./?.lua;./?/init.lua;
+-lua_package_cpath = NONE
++lua_package_path = ./?.lua;./?/init.lua;/opt/homebrew/share/lua/5.1/?.lua;/opt/homebrew/share/lua/5.1/?/init.lua
++lua_package_cpath = /opt/homebrew/lib/lua/5.1/?.so;
```
Comment (fffonion) [inline Formula/kong.rb:78]: let's use `;;` at the end, so we are adding new search path but not deleting old ones. same aplies to lua_package_path

File: Formula/kong.rb (hunk 1 above)
Comment (gszr) [reply Formula/kong.rb:67]: Homebrew does a replacement before applying the patch; see the end of section https://docs.brew.sh/Formula-Cookbook#patches. (Took me a while to find this btw 😅)

Comment (fffonion) [reply Formula/kong.rb:67]: that's hacky but i like it 😄

**Review by fffonion — APPROVED** [review]

**Comment by gszr** [comment]
I will run this once again in an m1 environment then we can merge it. Thanks for reviewing, @fffonion!

**Pull request merged**